united.py

```diff
@@ -28,10 +54,224 @@ def parse_date(created):
+        temp_dis = math.sqrt(lat_difference + lon_difference)
+        if temp_dis < min_distance:
+            min_distance = temp_dis
+            station_place_in_xml = i
+
+    return station_place_in_xml
+
+
+def convert_xml_values_to_numbers(rain):
+    num_conv = rain[:2]  # variable to help convert from string to number
+
+    for char in num_conv:  # in the xml number are in a three digits format (4-004), we delete the 0es before the number
+        if char == '0':
+            rain.replace(char, '')
+        else:
+            break
+
+    rain_in_millimeters = float(rain)
+
+    if rain_in_millimeters >= 990:
+        # numbers that are higher then 990 in the xml code equals 0.(the last digit) for example 991 = 0.1
+        rain_in_millimeters *= 0.01
+        # str_rain = "0." + rain
+        # rain_in_millimeters = float(str_rain)  # convert string to numbers
+
+    return rain_in_millimeters
+
+
+def convert_xml_numbers_to_hours(rain_duration):
+    # convert the xml code for number of hours, to the actual number of hours
+    str_duration = str(rain_duration)
+    str_duration = str_duration.strip(' ')
+    hours = {"1": 6, "2": 12, "3": 18, "4": 24, "/": 24, "5": 1, "6": 2, "7": 3, "8": 9, "9": 15}
+
+    return hours[str_duration
+
+
+    ]
+
+
+def convert_xml_numbers_to_current_weather(weather_code):
+    print weather_code
+    str_weather = str(weather_code)
+    str_weather = str_weather.strip(' ')
+    weather = {"0": 1, "1": 2, "3": 3, "4": 4, "5": 5, "7": 6, "8": 6, "9": 7, "10": 8, "11": 9,
+               "12": 10, "17": 11, "18": 12, "19": 13, "20": 14, "21": 15, "22": 16, "23": 17, "24": 18,
+               "25": 19, "26": 20, "27": 21, "28": 22, "29": 23, "30": 24, "31": 24, "32": 24, "33": 7,
+               "34": 7, "35": 7, "36": 25, "37": 25, "38": 25, "39": 25, "40": 26, "41": 27, "42": 28,
+               "43": 29, "44": 9, "45": 30, "46": 30, "47": 30, "48": 31, "49": 32, "50": 33, "51": 34,
+               "52": 33, "53": 35, "54": 36, "55": 37, "56": 38, "57": 39, "58": 37, "59": 37, "61": 37, "60": 36,
+               "62": 40, "63": 15, "64": 41, "65": 19, "66": 42, "67": 43, "68": 44, "69": 45, "70": 46, "71": 47,
+               "72": 48, "73": 16, "74": 50, "75": 51, "76": 52, "77": 53, "78": 54, "79": 55, "80": 56, "81": 57,
+               "82": 58, "83": 59, "84": 60, "85": 61, "86": 62, "87": 63, "88": 64, "89": 65, "90": 66, "91": 67,
+               "92": 68, "93": 69, "94": 70, "95": 71, "96": 72, "97": 73, "98": 74, "99": 75}
+
+    return weather[str_weather]
+
+
+def process_weather_data(latitude, longitude):
+
+    weather = 1  # default weather is clear sky
+    rain_in_millimeters = 0
+    rain_hours = 0
+
+    station = find_station_by_coordinate(latitude, longitude)
+    weather_data = collection.getElementsByTagName('surface_observation')
+
+    wind_force = weather_data[station].getElementsByTagName('FF')
```

Comment:
Since you keep doing the same thing but with a different tag string, perhaps extract this to a function